agent/src/agent/pipeline/config/js_scripts/protocol_3.js

```diff
@@ -19,17 +19,13 @@ function replace_illegal_chars(str) {
 
 function get_dimensions(record) {
     var dimensions = {};
-    for (var j = 0; j < state['DIMENSIONS'].length; j++) {
-        var dimension = extract_value(record.value, replace_illegal_chars(state['DIMENSIONS'][j]))
-        if (dimension === null) {
-            continue;
-        }
-        dimension = String(dimension).trim()
-        if (dimension === '') {
-            continue;
+    for (var dimension_path in state['DIMENSIONS']) {
```

Comment:
a client specifies in the config 'new name': 'original path', but here its the other way around, I think it's confusing, in the client config you should also have 'original path': 'new name'